frontend/src/HOC/withDataFetching.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useState } from 'react';
+import Style from './withDataFetching.style';
+
+const withDataFetching = <Props extends object>(
+  dataName: string,
+  fetchFunction: (props: Props) => any,
+  shouldCallEffect: (props: Props) => any[],
+) => (BaseComponent: React.ComponentType<Props>) => (props: Props) => {
+  const [loading, setLoading] = useState<boolean>(false);
+  const [data, setData] = useState<any>();
+  const [error, setError] = useState<string | null>();
+  useEffect(() => {
+    const fetch = async () => {
+      setLoading(true);
+      try {
+        const { body: fetchedData } = await fetchFunction(props);
+        setData(fetchedData);
+      } catch (caughtError) {
+        setError(caughtError.toString());
+      }
+      setLoading(false);
+    };
+    fetch();
+  }, [...shouldCallEffect(props)]);
```

Comment:
> ./src/HOC/withDataFetching.tsx
  Line 24:  React Hook useEffect has a missing dependency: 'props'. Either include it or remove the dependency array                                                 react-hooks/exhaustive-deps
  Line 24:  React Hook useEffect has a spread element in its dependency array. This means we can't statically verify whether you've passed the correct dependencies  react-hooks/exhaustive-deps

I have those warnings caused by this line, can they be resolved ?

Reply:
As indicated here https://fr.reactjs.org/docs/hooks-rules.html#only-call-hooks-from-react-functions
Hook should be used only in pure React Component Function in order to avoid errors.

An quick and dirty solution would be adding eslint comment // eslint-disable-next-line to the line before

Elsewhere, I found out this solution :
Update your HOC in order that your hook only use properties of the current React component :
```
const withDataFetching = <Props extends object>(BaseComponent: React.ComponentType<Props>) => (
  props: any,
) => {
  const [loading, setLoading] = useState<boolean>(false);
  const [data, setData] = useState<any>();
  const [error, setError] = useState<string | null>();
  const { shouldCallEffect, fetchFunction } = props;
  useEffect(() => {
    const fetch = async () => {
      setLoading(true);
      try {
        const { body: fetchedData } = await fetchFunction();
        setData(fetchedData);
      } catch (caughtError) {
        setError(caughtError.toString());
      }
      setLoading(false);
    };
    fetch();
  }, [shouldCallEffect, fetchFunction]);
  return (
    <Style.Container>
```

And then you can call it like below : 

```
const HomeWithDataFetching = withDataFetching(Home);

export default (props: any) => (
  <HomeWithDataFetching
    {...props}
    dataName="pokemons"
    fetchFunction={() => makeGetRequest(`/pokemon?page=${props.match.params.page}`)}
    shouldCallEffect={[props.match.params.page]}
  />
);
```